apps/system/js/app_chrome.js

```diff
@@ -709,6 +710,16 @@
 
       this.updateAddToHomeButton();
 
+      // We only change the icon back to the default one if the new page is not
+      // on the same domain than the previous one.
+      // In both cases, we look for the best icon after `mozbrowserloadend`.
+      var origin = this.getOriginFromURL(evt.detail);
```

Comment:
My understanding is that the new URL() approach performs better than creating a whole node object. Is there a simple way to work around the //example.com by prepending the scheme from a base URL somehow do you think?